enhancements/proposals/1277-expose-scyllacluster/README.md

```diff
@@ -0,0 +1,398 @@
+ensuring that the database cluster can seamlessly communicate with external systems and clients, while also offering the
+ability to fine-tune the communication settings based on specific traffic requirements.
+
+### Goals
+
+* Proposing API for new switch allowing different types of network configuration.
+* Describe use cases for different types of network configurations.
+* Describe how it affects existing procedures using ClusterIP as node identifiers and provide alignments.
+
+### Non-Goals
+
+Cover all possible network configurations.
+
+## Proposal
+
+I propose to extend the ScyllaCluster API with new fields adding a switch which will control how nodes of ScyllaDB
+broadcast their IP address to their clients and other nodes and on which interfaces they are listening. 
+Depending on the value of the switch, nodes may be accessible only privately, internally on the same VPC, externally 
+to the public Internet. Operator will be responsible for reconciling required resources and preparing the right ScyllaDB 
+configuration. Existing procedures relying on node IP identifiers will be aligned to these changes.
+
+### Risks and Mitigations
+
+#### ScyllaDB version requirement
+
+Node replacement procedure alignment requires the newest 5.2 ScyllaDB Open Source and 2023.1 Enterprise, meaning node
+replacement procedure won’t work when Scylla doesn’t meet minimum version criteria. Users wanting to update to a newer
+Operator version will need to update their ScyllaDB versions. We will mention it in release notes.
+
+## Design Details
+
+### Network connectivity cases
+
+#### Private Kubernetes
+
+This is the current setup type (Scylla Operator <= 1.10), only Pods deployed in the same K8s cluster can connect to
+Scylla Pods. MultiDC is not possible, as Service ClusterIP are virtual IPs which are not routable outside the k8s
+cluster.
+
+![Private-K8s](private-k8s.svg "Private K8s")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: ClusterIP
+broadcast_rpc_address: ClusterIP
+```
+
+---
+
+#### Private VPC
+
+In Private VPC type, Scylla Pods broadcast their ephemeral PodIP both to clients and other nodes. Because PodIPs are
+natively routable within a VPC, MultiDC is possible over private IPs. Public Internet access is not possible.
+This assumes Scylla supports UUID Node Identifiers and ephemeral IPs.
+Scylla Operator is configured to create Headless Services.
+
+![Private-VPC](private-vpc.svg "Private VPC")
+
+##### ScyllaDB configuration
+
+```yaml
+broadcast_address: PodIP
+broadcast_rpc_address: PodIP
+```
+
+---
+
+#### Private VPC + Public LB
+
+Scylla Nodes talk to each other over private IP, but all the clients, no matter where they are deployed, need to talk
+through the public Internet. MultiDC is possible over private IPs.
+Big disadvantage of this type is the cost associated with the necessity of going through the public Internet, and the
```

Comment:
Nodes broadcast address from the Internet network. In the picture every client crosses Internet boundary, it's in sync how it would behave without any smart routing.